inspire_matcher/core.py

```diff
@@ -192,42 +192,46 @@ def _compile_nested(query, record):
             },
         },
     }
+    return nested_query, paths, search_paths
+
 
+def _compile_nested(query, record):
+    nested_query, paths, search_paths = _create_nested_template(query)
     for path, search_path in zip(paths, search_paths):
         value = get_value(record, path)
         if not value:
             return
 
-        result['query']['nested']['query']['bool']['must'].append({
+        nested_query['query']['nested']['query']['bool']['must'].append({
             'match': {
                 search_path: value,
             },
         })
 
-    return result
+    return nested_query
 
 
 def _compile_nested_prefix(query, record):
-    path, search_path = query['path'], query['search_path']
-    value = get_value(record, path)
-    if not value:
-        return
-
-    nested_path = search_path.split('.')[0]
-    result = {
-        "query": {
-            "nested": {
-                "path": nested_path,
-                "query": {
-                    'match_phrase_prefix': {
-                        search_path: value,
-                    }
+    nested_query, paths, search_paths = _create_nested_template(query)
+    prefix_field = query['prefix_search_path'] if 'prefix_search_path' in query else None
```

Comment:
```suggestion
    prefix_field = query.get('prefix_search_path')
```
☝️ This will return `None` if the key doesn't exist

Reply:
or you could just `get('whatever', [])` and don't check on line `221`